lib/jxl/modular/encoding/enc_encoding.cc

```diff
@@ -237,6 +237,25 @@ Status EncodeModularChannelMAANS(const Image &image, pixel_type chan,
         wp_state.UpdateErrors(r[x], x, y, channel.w);
       }
     }
+  } else if (tree.size() == 1 && tree[0].predictor == Predictor::Gradient &&
```

Comment:
Can we add a test that exercises this path and one that does but uses skip_encoder_fast_path?

Reply:
There actually isn't a specific decoder fast path for this (yet) — it goes to the slightly more generic decoder path of "no tree, not WP". Do you think a test would be useful here?

Reply:
Are you sure there isn't? I remember adding one...

A test would be useful anyway :)

Reply:
There is a decoder path for `is_gradient_only`, actually that's the one that will be used here, with an unnecessary ctx lookup because everything has the same ctx. Maybe I should add a decoder path for this no-tree gradient-only no-multiply/offset case, it might be slightly faster.

There's no `skip_decoder_fast_path`, right? Maybe it would be useful to also test that — encode with fast paths, decode with generic path...

Reply:
Oh, this is single-tree gradient only... I think that's OK as it is on the decoder side. Still needs a test though :)

As for testing that - likely not needed if we test the encoder fast paths...

Reply:
Changing the decoder fast paths to first see if it's a no-tree case before it tries to do the lookup table thing for WP/Gradient.

Adding a specific decode fast path for no-tree gradient-only helps a bit.

before:
4064 x 2704, geomean: 47.08 MP/s [37.02, 49.18], 30 reps, 4 threads.
after:
4064 x 2704, geomean: 52.10 MP/s [42.82, 54.56], 30 reps, 4 threads.

Added a test a for the encoder skip_fast_path for lightning mode.

Reply:
Made the above changes, PTAL.